screens/Matches.js

```diff
@@ -1,23 +1,139 @@
-import React from 'react';
-import styles from '../styles'
+import React from "react";
+import styles from "../styles";
+import { connect } from "react-redux";
+import * as firebase from "firebase";
 
-import { 
-  Text, 
-  View
-} from 'react-native';
+import { Text, ScrollView, TouchableOpacity, Image, View } from "react-native";
 
 class Matches extends React.Component {
-  state = {}
+  constructor() {
+    super();
+    this.roomref = firebase.firestore().collection("room");
+    this.userref = firebase.firestore().collection("users");
+    this.state = {
+      isLoading: true,
+      userid1: "",
+      userid2: "",
+      key: "",
+      swipes: [],
+      swipes1: [],
+      swipes2: []
+    };
+  }
+
+  usersinroom = roomref => {
+    const room = firebase
+      .firestore()
+      .collection("room")
+      .doc(roomref);
+    return room.get().then(doc => {
+      if (doc.exists) {
+        userid1: "";
+        users = doc.data();
+        this.setState({
+          userid1: users.userid1,
+          userid2: users.userid2
+        });
+      } else {
+        console.log("No such document!");
+      }
+    });
+  };
+
+  swipesyes = userid => {
+    return this.userref
+      .doc(userid)
+      .get()
+      .then(doc => {
+        if (doc.exists) {
+          const user = doc.data();
+          this.setState({
+            swipes: user.swipesyes
+          });
+        } else {
+          console.log("No such document! 2");
+        }
+      });
+  };
+
+  cardsdetails = idlist => {
+    var cardsref = firebase.firestore().collection("cards");
+    const boards = [];
+    idlist.forEach(function(element) {
```

Comment:
Du coup, pour que cette fonction retourne une promesse:

- je vois que tu as une liste d'opérations asynchrones. Dans ce cas, on utilise `Promise.all()` et on fournit en paramètre une liste de promesses.
- du coup si tu utilises `map` au lieu de `forEach` tu vas bien récupérer un tableau de promesses (car `return cardsref.doc().blabla()` retourne une promesse à chaque itération)
- et derrière tu peux utiliser `Promise.all` avec le résultat